### DISCUSSION TITLE
Kalkulator opłat prowizji

### DISCUSSION THREAD

**Discussion by Eskim83** [allegro/allegro-api#7768] [Suggestion - verification]
Aktualnie, aby sprawdzić prowizję dla aukcji należy najpierw pobrać obiekt auction z serwera, a następnie go przekazać w odpowiednim parametrze. Weryfikacja kosztów dla kilkudziesięciu tysięcy aukcji jest czasochłonna i zasobożerna. Ponadto mam wrażenie, że nie są uwzględniane promocje na produkcie (np. Allegro Ceny czy Strefa Okazji).
Posiadając już podstawowe informacje o produktach (cena, kategoria itp.) do niedawna wystarczyło je przekazać w metodzie [POST /pricing/offer-fee-preview](https://developer.allegro.pl/documentation/#operation/calculateFeePreviewUsingPOST). Aktualnie wymagane jest przekazanie parametrów oferty co wymusza pobieranie samej oferty.

Dodatkowo, aby wystawiać świadomie sprawdzaliśmy programowo koszty wystawienia w zadanej kategorii. Aktualnie trzeba konstruować pseudoaukcję (jeszcze nie wiem jak to zrobić w związku z wymaganym elementem parameters).

Zamiast upraszczać, zostało to skomplikowane, a wystarczyłyby dwie metody (albo jedna)):
* pobierająca koszty dla wskazanej aukcji po id oferty (prowizja przy sprzedaży, prowizja, prowizja z wyróżnieniem itd. - chcę wiedzieć ile zapłacimy jak włączymy np. wyróżnienie, ale też chcę wiedzieć ile zapłacimy jak sprzedamy produkt) 
* pobierająca koszty sprzedaży tak jak to robi kalkulator opłat - dla kategorii (tak to działało wcześniej - podawałem m.in. kategorię, cenę i otrzymywałem w odpowiedzi koszty - mam listing aktywnych aukcji w bazie i znam kategorię, cenę itd.)

Przede wszystkim chcemy wiedzieć:
* ile zapłacimy zanim sprzedamy produkt?
* ile będzie nas kosztowało wystawienie produktu (wystawienie + ew. koszty sprzedaży)?
* o ile więcej zapłacimy za sprzedaż jak wyróżnimy ofertę?

Świadome działania w zakresie kontroli ceny są podstawowym elementem pracy na Allegro każdego dnia.

Aktualna zmiana spowodowała opóźnienia w zakresie pobierania zamówień i wielu innych procesów, gdyż musiałem do szybkiej metody pobierającej dane tak jak to robi kalkulator dodać pobieranie aukcji (dodać też sam numer aukcji, który nie był wcale potrzebny, bo wystarczyła kategoria i cen). Sama weryfikacja kosztów odbywa się u nas na wielu płaszczyznach. Oczywiście będę się to starał zoptymalizować, aby wykluczyć pobieranie oferty, ale trudno symulować element "parameters".. tym bardziej, że nie znam jeszcze dokładnych wymagań tego elementu.

**Comment by EmilianNowak** [comment]
Dziękujemy za przesłane uwagi, postaramy się je uwzględnić przy kolejnej wersji API.

Odnośnie elementu parameters - nie jest on wymagalny, można podać pustą tablicę
```
"parameters": []
```

Jednak jest jeden parametr, który dla niektórych opłat wpływa na cenę. Jest to stan przedmiotu w ofercie: NOWY/UŻYWANY.
Stan ma kod: 11323, wartość na NOWY=11323_1, UŻYWANY=11323_2.

Czyli aby zasymulować używany przedmiot (bez konieczności zapisania i pobrania oferty) można podać:
```json
   "parameters": [
      {
        "id": "11323",
        "valuesIds": [
          "11323_2"
        ]
      }
   ]
```
